src/lua/digest.lua

```diff
@@ -267,4 +313,70 @@ m['aes256cbc'] = {
     end
 }
 
+for _, var in ipairs({'32', '64'}) do
+    local xxHash
+
+    local update_fn_name = 'XXH' .. var .. '_update'
+    local digest_fn_name = 'XXH' .. var .. '_digest'
+    local reset_fn_name = 'XXH' .. var .. '_reset'
+    local copy_fn_name = 'XXH' .. var .. '_copyState'
+    local methods = {
+        default_seed = 0,
+
+        update = function(self, str)
+            if type(str) ~= 'string' then
+                error("Usage xxhash" .. var .. ":update(string)")
+            end
+            return builtin[update_fn_name](self, str, #str) == builtin.XXH_OK
+        end,
+        result = function(self)
+            return builtin[digest_fn_name](self)
+        end,
+        clear = function(self, seed)
+            if seed == nil then
+                seed = self.default_seed
+            end
+            return builtin[reset_fn_name](self, seed) == builtin.XXH_OK
+        end,
+        copy = function(self)
+            local copy = xxHash.new()
+            builtin[copy_fn_name](copy, self)
+            return copy
+        end,
+    }
+
+    local XXH_state_t = ffi.typeof('XXH' .. var .. '_state_t')
+    ffi.metatype(XXH_state_t, {
+        __index = function(_, key)
+            return methods[key]
+        end,
+        __tostring = function(self)
+            return string.format('value: %s', self:result())
+        end,
+    })
+
+    xxHash = {
+        new = function(seed)
+            local value = ffi.new(XXH_state_t)
+            value:clear(seed)
+            return value
```

Comment:
From what I see in pmur code, `default_seed` is stored as an attribute so as to update it in `new(seed)` in case `seed` is specified. Maybe you need to do the same.

Reply:
Well, I add "default_seed" attribute to `digest.xxhash.new()` table.

Reply:
As far as I understand, "0" in these two places are related values. Maybe you should use an attribute for this?
```
 default_seed = seed or 0,
```
```
 if seed == nil then
    seed = 0
end
```

Reply:
I inlined it as Vlad suggested. See comment - https://github.com/tarantool/tarantool/pull/5828#discussion_r593508162